packages/gamut-labs/src/ListSection/index.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { ReactElement, useState } from 'react';
+
+import { PageSection } from '..';
+
+export type ListSectionProps<T> = {
+  title: string;
+  /**
+   * Array of elements.
+   */
+  listItems: T[];
+  /**
+   * Render function that will render each element in the list items.
+   */
+  renderItem: (item: T, index?: number) => ReactElement;
+  /**
+   * Number of items to be initally displayed (note: if the initial display amount matches the number of list items, Show All button will not appear).
+   */
+  initialDisplayAmount?: number;
+};
+
+export const ListSection = <T extends unknown>({
+  title,
+  listItems,
+  renderItem,
+  initialDisplayAmount = 3,
+}: ListSectionProps<T>) => {
+  const [showAll, setShowAll] = useState(false);
+
+  const handleShowAll = () => {
+    setShowAll(!showAll);
+  };
+
+  const renderItems = () => {
+    const displayLimit = showAll ? listItems.length : initialDisplayAmount;
+    return listItems.slice(0, displayLimit).map((item) => renderItem(item));
+  };
+
+  const renderfooterButton = () => {
```

Comment:
Nit: capitalization?

```suggestion
  const renderFooterButton = () => {
```

This is part of why I personally prefer not to use inline `render*` functions; IMO they add a bit of extra overhead. Not a blocker from me, just a side comment.

Reply:
Fixed the capitalization! Also agree that they are not always the best and can add extra overhead. I go back and forth but I think it works here